tank/alexnet_torch/alexnet_torch_test.py

```diff
@@ -57,49 +59,19 @@ def configure(self, pytestconfig):
         self.module_tester = AlexnetModuleTester(self)
         self.module_tester.benchmark = pytestconfig.getoption("benchmark")
 
-    def test_module_static_cpu(self):
-        dynamic = False
-        device = "cpu"
-        self.module_tester.create_and_check_module(dynamic, device)
-
-    def test_module_dynamic_cpu(self):
-        dynamic = True
-        device = "cpu"
-        self.module_tester.create_and_check_module(dynamic, device)
-
-    @pytest.mark.skipif(
-        check_device_drivers("gpu"), reason=device_driver_info("gpu")
-    )
-    def test_module_static_gpu(self):
-        dynamic = False
-        device = "gpu"
-        self.module_tester.create_and_check_module(dynamic, device)
-
-    @pytest.mark.skipif(
-        check_device_drivers("gpu"), reason=device_driver_info("gpu")
-    )
-    def test_module_dynamic_gpu(self):
-        dynamic = True
-        device = "gpu"
-        self.module_tester.create_and_check_module(dynamic, device)
-
-    @pytest.mark.skipif(
-        check_device_drivers("vulkan"), reason=device_driver_info("vulkan")
-    )
-    @pytest.mark.xfail(
-        reason="Issue known, WIP",
-    )
-    def test_module_static_vulkan(self):
-        dynamic = False
-        device = "vulkan"
-        self.module_tester.create_and_check_module(dynamic, device)
-
-    @pytest.mark.skipif(
-        check_device_drivers("vulkan"), reason=device_driver_info("vulkan")
-    )
-    def test_module_dynamic_vulkan(self):
-        dynamic = True
-        device = "vulkan"
+    param_list = get_valid_test_params()
+
+    @parameterized.expand(param_list, name_func=shark_test_name_func)
+    def test_module(self, dynamic, device):
+        if device == "gpu":
+            if check_device_drivers("gpu"):
+                pytest.skip(reason=device_driver_info("gpu"))
+        elif device == "vulkan":
+            if check_device_drivers("vulkan"):
+                pytest.skip(reason=device_driver_info("vulkan"))
+            else:
+                if dynamic == False:
```

Comment:
although simple, is this the cleanest way we can use? I was thinking we can have some dict like data structure where it’s easy for developer to add or mark xfail cases, any thoughts @monorimet

Reply:
I'm adding support for this in a patch i'm working on

Reply:
Looks like Dan has some ideas about this. 
LGTM as long as we can xfail easily.

Reply:
It's also important to be able to easily designate a specific test case or a subset of test cases based on model, dynamic, device params.
We've been doing `pytest tank/albert-base-v2_torch/ -k "static_gpu"` , for example. As long as we can do that just as easily this looks great! :)

Reply:
> It's also important to be able to easily designate a specific test case or a subset of test cases based on model, dynamic, device params. We've been doing `pytest tank/albert-base-v2_torch/ -k "static_gpu"` , for example. As long as we can do that just as easily this looks great! :)

This should not change as the final test cases will have same names as earlier, here we're eliminating the need to explicitly write all the test cases.

Reply:
Will go ahead with this for now and make the required changes later on.